appinventor/components/src/com/google/appinventor/components/runtime/Form.java

```diff
@@ -682,11 +684,20 @@ void ReplayFormOrientation() {
     // We first make a copy of the existing dimChanges list
     // because while we are replaying it, it is being appended to
     Log.d(LOG_TAG, "ReplayFormOrientation()");
-    ArrayList<PercentStorageRecord> temp = (ArrayList<PercentStorageRecord>) dimChanges.clone();
+    LinkedHashMap<Integer, PercentStorageRecord> temp = (LinkedHashMap<Integer, PercentStorageRecord>) dimChanges.clone();
     dimChanges.clear();         // Empties it out
-    for (int i = 0; i < temp.size(); i++) {
-      // Iterate over the list...
-      PercentStorageRecord r = temp.get(i);
+    // for (int i = 0; i < temp.size(); i++) {
+    //   // Iterate over the list...
+    //   PercentStorageRecord r = temp.get(i);
+    //   if (r.dim == PercentStorageRecord.Dim.HEIGHT) {
+    //     r.component.Height(r.length);
+    //   } else {
+    //     r.component.Width(r.length);
+    //   }
+    // }
+    // Iterate temp
+    for (Integer key : temp.keySet()) {
```

Comment:
Rather than iterate over `keySet()`, iterate over `values()`. It will save having to do the get on the next line.